core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -3406,9 +3406,23 @@ abstract Expression implementSafe(RexToLixTranslator translator,
       translatedOperands.add(wmColExpr);
       translatedOperands.add(intervalExpression);
 
-      return Expressions.call(BuiltInMethod.TUMBLING.method, inputEnumerable,
-          EnumUtils.tumblingWindowSelector(inputPhysType, outputPhysType,
-              translatedOperands.get(0), translatedOperands.get(1)));
+      // handle the optional offset parameter.
+      if (call.getOperands().size() > 2) {
+        translatedOperands.add(translator.translate(call.getOperands().get(2)));
+      } else {
+        // use 0 for the default value when offset parameter is not set.
```

Comment:
Can we simpilfy this logic as below? It may be more intuitive.
```
// handle the optional offset parameter, use 0 as default when ...
final Expression xxxExpression = call.getOperands().size() > 2 ? translator.translate(call.getOperands().get(2)) : Expressions.constant(0, long.class);
translatedOperands.add(xxxExpression);
```

Reply:
Done